cpp/util/diagonalize.cc

```diff
@@ -2,6 +2,14 @@
 
 namespace freud { namespace util {
 
+void diagonalize33SymmetricMatrix(const util::ManagedArray<float> &mat, float eigen_vals[3], float eigen_vecs[9])
```

Comment:
Does this act the same as if you passed `&mat[0]` (or some raw pointer to the first element) to the other constructor?

Reply:
Yes, it would, but currently the only reason I didn't change the function altogether is because it's also used in LocalDescriptors. Once I update that as well, I will remove the old function signature here. I'm trying to remove references to raw arrays as much as possible so that everything can use `ManagedArray`. We may at some point want to update `ManagedArray` to support stack allocated arrays for certain specific fixed sizes for convenience, though, but I don't think that's urgent.